vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridMultiSelectionColumnPageIT.java

```diff
@@ -164,4 +181,16 @@ public void setAutoWidthOfSelectionColumnIsTrue() {
         String autoWidth = gridSelectionMode.getAttribute("autoWidth");
         Assert.assertTrue("autoWidth should be true", Boolean.parseBoolean(autoWidth));
     }
+
+    @Test // https://github.com/vaadin/vaadin-flow-components/issues/411
+    public void selectRow_gridWithUnknownItemCount_noExceptionThrown() {
+        open();
+        GridElement grid = $(GridElement.class).id(
+                GridMultiSelectionColumnPage.UNKNOWN_ITEM_COUNT_LAZY_GRID_ID);
+        // Select any row
+        grid.findElements(By.tagName("vaadin-checkbox")).get(0).click();
```

Comment:
Just for fun, maybe check that scrolling to an item outside the initial range and selecting there works too ?
And maybe test unselecting too even though it should not matter

Reply:
Done